realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -481,6 +495,17 @@ public Builder rxFactory(RxObservableFactory factory) {
             return this;
         }
 
+        /**
+         * Sets the initial data in {@link io.realm.Realm}. This transaction will be executed only for the first time
+         * when database file is created or while migrating the data when {@link Builder#deleteRealmIfMigrationNeeded()} is set.
+         *
+         * @param transaction transaction to execute.
+         */
+        public Builder initialData(Realm.Transaction transaction) {
```

Comment:
Note that a RealmConfiguration can be used to create both an `DynamicRealm` and a typed `Realm`. The issue didn't really describe this, but it means we probably need to support both 

```
public Builder initialData(Realm.Transaction transaction);
public Builder initialData(DynamicRealm.Transaction transaction);
```

It is slightly annoying though as the semantics gets a bit muddled (e.g. opening a DynamicRealm will not trigger the typed transaction and vice versa), but it might be the best of multiple evils.

What do you think @thesurix @realm/java ?